samples/display/lvgl/sample.gui.lvgl fails to build on several boards

Doing:

`./scripts/sanitycheck -s samples/display/lvgl/sample.gui.lvgl -l`

I get:

```
ERROR   - mimxrt1064_evk            samples/display/lvgl/sample.gui.lvgl               FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/mimxrt1064_evk/samples/display/lvgl/sample.gui.lvgl/build.log
INFO    - Total complete:    1/   4  25%  skipped:    0, failed:    1
ERROR   - mimxrt1060_evk            samples/display/lvgl/sample.gui.lvgl               FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/mimxrt1060_evk/samples/display/lvgl/sample.gui.lvgl/build.log
INFO    - Total complete:    2/   4  50%  skipped:    0, failed:    2
ERROR   - mimxrt1050_evk            samples/display/lvgl/sample.gui.lvgl               FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/mimxrt1050_evk/samples/display/lvgl/sample.gui.lvgl/build.log
INFO    - Total complete:    3/   4  75%  skipped:    0, failed:    3
ERROR   - reel_board                samples/display/lvgl/sample.gui.lvgl               FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/reel_board/samples/display/lvgl/sample.gui.lvgl/build.log
```

This seems due to:

```
[ 71%] Building C object modules/lvgl/CMakeFiles/..__modules__lib__gui__lvgl.dir/src/lv_font/lv_font_dejavu_16_persian_hebrew.c.obj
[ 71%] Building C object modules/lvgl/CMakeFiles/..__modules__lib__gui__lvgl.dir/src/lv_font/lv_font_fmt_txt.c.obj
/home/galak/git/modules/lib/gui/lvgl/src/lv_font/lv_font_fmt_txt.c:349:13: error: 'decompress' defined but not used [-Werror=unused-function]
  349 | static void decompress(const uint8_t * in, uint8_t * out, lv_coord_t w, lv_coord_t h, uint8_t bpp, bool prefilter)
      |             ^~~~~~~~~~
cc1: all warnings being treated as errors
gmake[2]: *** [modules/lvgl/CMakeFiles/..__modules__lib__gui__lvgl.dir/build.make:603: modules/lvgl/CMakeFiles/..__modules__lib__gui__lvgl.dir/src/lv_font/lv_font_fmt_txt.c.obj] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:3275: modules/lvgl/CMakeFiles/..__modules__lib__gui__lvgl.dir/all] Error 2
gmake: *** [Makefile:104: all] Error 2
```

--- a/west.yml
+++ b/west.yml
@@ -87,7 +87,7 @@ manifest:
       revision: 7deb3aab1cc430a7650143f9833cfc8ddad76eb5
       path: modules/hal/quicklogic
     - name: lvgl
-      revision: 9339a1331cc36e82c180d8277ab22953282595d9
+      revision: 31acbaa36e9e74ab88ac81e3d21e7f1d00a71136
       path: modules/lib/gui/lvgl
     - name: mbedtls
       revision: e3e135f89693a0342bbc72f31a4e00cdf36c3977

